### DISCUSSION TITLE
How to change Elsa Studio(v3) HttpEndpoint Activity BaseUrl?

### DISCUSSION THREAD

**Discussion by sagegz** [elsa-workflows/elsa-core#5191] [Q&A]
I searched a lot about how to change Elsa Studio Http Endpoint Activity's BaseUrl but got nothing about it.
Could you please give me an instruction in detail?
1. Docker (http://localhost:13000/api/workflows)
2. Elsa Studio Blazor Server(http://localhost/workflows)
![HttpEndpoint](https://github.com/elsa-workflows/elsa-core/assets/56223336/e7f93844-8698-4ff0-83ac-f1526eb8ead3)

**Comment by glime-ai** [comment]
To change the base URL for the HTTP Endpoint activity in Elsa Studio, you would typically need to adjust the configuration settings that the Elsa services use when they start up. Here's how you can do it for both Docker and Elsa Studio Blazor Server:

1. **Docker**:
   When running Elsa with Docker, you can pass environment variables to the container to configure the base URL. For example, if you want to set the base URL to `http://localhost:13000/api/workflows`, you can pass the `ELSA__SERVER__BASEURL` environment variable when you start the container:

   ```shell
   docker run -t -i -e ELSA__SERVER__BASEURL=http://localhost:13000/api/workflows -p 13000:80 elsaworkflows/elsa-server-v3:latest
   ```

   Make sure to replace `elsaworkflows/elsa-server-v3:latest` with the appropriate Docker image tag for your setup.

2. **Elsa Studio Blazor Server**:
   For an Elsa Studio Blazor Server application, you would typically set the base URL in the `appsettings.json` file or through environment variables in your hosting environment. If you're running the application locally, you can modify the `appsettings.json` file to include the base URL:

   ```json
   {
     "Elsa": {
       "Server": {
         "BaseUrl": "http://localhost/workflows"
       }
     }
   }
   ```

   If you're setting it through environment variables, you can set `ELSA__SERVER__BASEURL` to the desired value.

Please note that the exact variable names and configuration methods may vary depending on the version of Elsa you are using and how your application is set up. If you are using a version of Elsa that does not recognize these environment variables or configuration settings, you may need to refer to the specific documentation for your version or seek support from the Elsa community.

Also, the screenshot URL you provided seems to be incorrect or incomplete, as it does not lead to an actual image. If you need further assistance with a specific part of the Elsa Studio interface, please provide a valid URL or a more detailed description of the issue.

**Comment by sagegz** [comment]
Thanks to AI.But I can't get the right answer after setting appsettings.json in situation 2.
`{
    "Logging": {
        "LogLevel": {
            "Default": "Information",
            "Microsoft.AspNetCore": "Warning"
        }
    },
    "AllowedHosts": "*",
    "Backend": {
        "Url": "https://localhost:5001/elsa/api"
    },
    "Elsa": {
        "Server": {
            "BaseUrl": "http://localhost:5001/workflows"
        }
    }
}`
It doesn't change anything. It's still http://localhost/workflows
![HttpEndpoint](https://github.com/elsa-workflows/elsa-core/assets/56223336/86d868e8-ea35-40c7-ac40-724dd606e01d)

**Comment by sagegz** [comment]
@sfmskywalker Can you give me some advice?

**Reply by sfmskywalker** [reply]
Try the following:

```
docker run -t -i -e ASPNETCORE_ENVIRONMENT=Development -e HTTP_PORTS=8080 -e HTTP__BASEURL=http://localhost:13000 -e HTTP__BASEPATH=/my-path-prefix -p 13000:8080 elsaworkflows/elsa-server-v3:latest
```

Notice the `HTTP__BASEPATH=/my-path-prefix` environment variable being passed in - that's what controls the path prefix for HTTP endpoints.

<img width="482" alt="image" src="https://github.com/elsa-workflows/elsa-core/assets/938393/6356057f-6ca8-4c82-8719-835bccaaefb9">

**Reply by sagegz** [reply]
Thanks for your reply. But I want to change the Port No. in Elsa Studio Blazor Server.It shows http://localhost/workflows. But I want to change like this  **http://localhost:5001/workflows** in official demo. @sfmskywalker
